test/gtest/multi_threading.cpp

```diff
@@ -156,8 +158,35 @@ TEST_F(MultiThreadingTestFixture, ConcurrentTransfers) {
     t2.join();
 }
 
+TEST_F(MultiThreadingTestFixture, GenerateNotification) {
+    nixlAgent local_agent = createAgent(local_agent_name);
+    nixlBackendH* backend = verifyMockDramBackendCreation(local_agent);
+    nixlAgent remote_agent = createAgent(remote_agent_name);
+    verifyMockDramBackendCreation(remote_agent);
+    nixl_opt_args_t extra_params = createExtraParams(backend);
+
+    verifyMemoryRegistration(local_agent, extra_params);
+
+    std::string md;
+    nixl_status_t status = remote_agent.getLocalMD(md);
+    EXPECT_EQ(status, NIXL_SUCCESS);
+    status = local_agent.loadRemoteMD(md, remote_agent_name);
+    EXPECT_EQ(status, NIXL_SUCCESS);
+
+    auto transfer_sequence = [&]() {
+        nixl_status_t status = local_agent.genNotif(remote_agent_name, "hello!");
+        EXPECT_EQ(status, NIXL_SUCCESS);
+    };
+
+    std::thread t1(transfer_sequence);
+    std::thread t2(transfer_sequence);
+
+    t1.join();
+    t2.join();
```

Comment:
Two things:
1. I think that to be meaningful, this test must check the consistency of the received notifications
2. Maybe we want to be aggressive with threads and spawn more than 2
3. Should we try to make the race more possible by (a) doing multiple iterations; (b) synchronizing the beginning of notification generation

Reply:
I think this was discussed already when introducing the thread sanitizer tests initially - they don't require neither more threads nor more iterations. If there is unsafe access to memory, then tsan will catch it on first iterations with two threads.

Reply:
Regarding consistency - this test uses mock backend which doesn't actually pass any notifications between agents so there is nothing to verify. GenNotif() is used by nixl_test from multi-threaded scenario though (which I extended with multi-threading to be the smoke test).